### ISSUE TITLE
How to propagate mouse events to graphical shell running the terminal?

### ISSUE THREAD

**Issue by least-olegs** [nsf/termbox-go#133]
Say, you are running in XTerm, or Konsole and you want to select text, right-click, use context menu to copy it and similar ops. How'd you do it?
My question implies that this doesn't work with demos / my minimal attempt. If it was meant to work, then I was trying this on Konsole and XTerm (there's no context menu on XTerm, but you can select text) .

**Comment by nsf** [comment]
There is no one way. Terminals don't really connected to X11 per se. And copy & paste buffer is a thing from X11. You can take a look at tmux for example how it implements copy & pasting in a terminal. So your best bet is to keep settings in a certain way in your terminal and then use selection and what not. Usually selecting text alone "copies" it to selection buffer. You can then paste it using middle mouse button.

To do proper copy & pasting you'll have to connect your app to X11 directly most likely.

**Comment by least-olegs** [comment]
Sorry, that's not exactly what I asked. I understand it that termbox "captures" input events (keyboard and mouse) to send them to the application using it.  This prevents the graphical shell from receiving those events (something that you would often want to do), but, eventually you just want to say "I'm not handling this event, please send it to whoever else was listening to it". For example, browser JavaScript can listen to text input events and decide to do something different with them, or it can decide to let the browser deal with them.

In other words, if I don't handle mouse click events, I would want the graphical shell hosting the terminal to handle it and do whatever that's shell action is assigned to that event.

**Comment by nsf** [comment]
Most terminals allows you to enable/disable mouse events and in termbox you can do that via https://godoc.org/github.com/nsf/termbox-go#SetInputMode, just don't set the InputMouse flag and termbox will inform the terminal that it doesn't want mouse events.

The rest is up to a terminal.
